src/time.cr

```diff
@@ -1254,6 +1270,17 @@ struct Time
     to_unix * 1_000 + (nanosecond // NANOSECONDS_PER_MILLISECOND)
   end
 
+  # Returns the number of nanoseconds since the Unix epoch
+  # (`1970-01-01 00:00:00.000000000 UTC`).
+  #
+  # ```
+  # time = Time.utc(2016, 1, 12, 3, 4, 5, nanosecond: 678_910_123)
+  # time.to_unix_ns # => 1452567845678910123
+  # ```
+  def to_unix_ns : Int64
+    (to_unix * NANOSECONDS_PER_SECOND) + nanosecond
```

Comment:
This causes an `OverflowError` for `Time.unix_ns(Int64::MIN).to_unix_ns`, because `nanosecond > 0` implies `to_unix * NANOSECONDS_PER_SECOND < Int64::MIN`.

This isn't a problem for `to_unix_ms` because `Int64`'s range far exceeds the representable range of milliseconds (the earliest being `Time.unix_ms(-62135596800000)`).